cloverage/src/cloverage/coverage.clj

```diff
@@ -288,16 +288,25 @@
                                                            :output  output
                                                            :forms   forms})))
 
-(defn- coverage-under? [forms failure-threshold]
-  (when (pos? failure-threshold)
-    (let [pct-covered (apply min (vals (rep/total-stats forms)))
-          failed?     (< pct-covered failure-threshold)]
-      (when failed?
-        (println "Failing build as coverage is below threshold of" failure-threshold "%"))
-      failed?)))
+(defn- coverage-under? [forms failure-threshold line-failure-threshold form-failure-threshold]
```

Comment:
I'd try adding test coverage to this defn - looks like there are a few possible combinations.

The 'precedence' should be particularly emphasized in the tests.